### PR TITLE
[DX-1628] Updated for 5.4

### PR DESCRIPTION
**Pull request by andyo-tyk**
DX-1628

---

### Preview Link
https://deploy-preview-5282--tyk-docs.netlify.app/docs/nightly/getting-started/using-oas-definitions/oas-reference/

### Description
Update to Tyk OAS supported-features matrix for features added in 5.4, no peer review required.

___

### **PR Type**
Documentation

___

### **Description**
- Updated the OpenAPI Specification (OAS) supported-features matrix to reflect new features and changes introduced in version 5.5.
- Improved consistency and clarity of feature names across the document.
- Added a new section for Traffic Logs, detailing features related to traffic logging and analytics.
- Enhanced documentation with additional details on existing features, improving user understanding.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>oas-reference.md</strong><dd><code>Update OAS supported-features matrix for version 5.5</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md

<li>Updated feature names for consistency and clarity.<br> <li> Added new features to the supported-features matrix for version 5.5.<br> <li> Introduced a new section for Traffic Logs with relevant features.<br> <li> Enhanced documentation with additional details on existing features.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5282/files#diff-740f4e9143de7a261a719a11fa17664649e5b883c4b6c250385b7800fe3f9341">+55/-56</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md [hunk 1]
```diff
@@ -47,46 +50,51 @@ If there's a feature you're looking to use that isn't yet implemented, let us kn
...
 | Feature                               | Implemented      | API Designer  |
 |---------------------------------------|------------------|---------------|
-| Detailed recording (in Log Browser)   | ✅               | ✅            |
-| Config Data                           | ✅               | ✅            |
-| Context Variables                     | ✅               | ✅            |
+| Custom config data                    | ✅               | ✅            |
+| Context variables                     | ✅               | ✅            |
 | CORS                                  | ✅               | ✅            |
-| Service Discovery                     | ✅               | ✅            |
-| Segment Tags                          | ✅               | ✅            |
```
Comment (dcs3spp) [inline tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:75]: Is segment tags now removed or has this been moved to API Management section?

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md [hunk 2]
```diff
@@ -47,46 +50,51 @@ If there's a feature you're looking to use that isn't yet implemented, let us kn
...
-| Upstream Request Signing              | ❌️               | ❌️            |
+| Upstream certificates (mTLS)          | ✅               | ✅            |
+| Public Key certificate pinning        | ✅               | ✅            |
+| GW-Upstream request signing           | ❌️               | ❌️            |
 
 ### API-level (Global) Features
 
 | Feature                               | Implemented      | API Designer  |
 |---------------------------------------|------------------|---------------|
-| Detailed recording (in Log Browser)   | ✅               | ✅            |
```
Comment (dcs3spp) [inline tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:70]: Has this been removed?

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md [hunk 3]
```diff
@@ -47,46 +50,51 @@ If there's a feature you're looking to use that isn't yet implemented, let us kn
...
-| Custom Response Plugin                    | ✅               | ✅            |
-| Custom Analytics Plugin                   | ❌️               | ❌️            |
-| Batch Requests                        | ❌️               | ❌️            |
-| Analytics API Tagging (tag_headers)   | ❌️               | ❌️            |
-| expire_analytics_after                | ❌️               | ❌️            |
-| Do not track Analytics (per API)      | ❌️               | ❌️            |
-| Webhooks                              | ❌️               | ❌️            |
-| Preserve Host Header                  | ❌️               | ❌️            |
-| Transport (proxy.transport)           | ❌️               | ❌️            |
+| Custom request plugins (pre/preAuth/post) | ✅               | ✅            |
```
Comment (dcs3spp) [inline tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:80]: Why are custom analytics plugin, analytics API tagging, expire_analytics_after and do not track analytics (per API), transport (proxy.transport) removed from table or have these being renamed and included in the new Traffic Logs section?

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md [hunk 4]
```diff
@@ -95,28 +103,19 @@ If there's a feature you're looking to use that isn't yet implemented, let us kn
...
-| Response Header Transform             | ✅               | ✅            |
-| Response Body Transform               | ✅               | ✅            |
-| Mock Response                         | ✅               | ✅            |
-| Virtual Endpoint                      | ✅               | ✅            |
-
-### Working with Tyk OAS API Definitions
-
-| Feature                               | Implemented      | API Designer  |
-|---------------------------------------|------------------|---------------|
-| API Export                            | ✅               | ✅            |
```
Comment (dcs3spp) [inline tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:119]: Why has API Export, API Raw OAS Editor, API Endpoint CRUD Operations, Middleware CRUD Operations being removed?

**Review by dcs3spp — COMMENTED** [review]
Made some comments to double check items moved between tables, once confirmed will merge and release. Should this be published to docs v5.4 only or docs v5.4 and v5.5? Does the PR relate to ticket DX-1628?
https://tyktech.atlassian.net/browse/DX-1628

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md (hunk 1 above)
Comment (andyo-tyk) [reply tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:75]: Moved

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md (hunk 2 above)
Comment (andyo-tyk) [reply tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:70]: Moved to the new Traffic Logs section

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md (hunk 3 above)
Comment (andyo-tyk) [reply tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:80]: Moved to the new Traffic Logs section

File: tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md (hunk 4 above)
Comment (andyo-tyk) [reply tyk-docs/content/getting-started/using-oas-definitions/oas-reference.md:119]: I don't think these are particularly relevant for customers to see in this list, so have removed them. They can all be assumed (or we wouldn't be encouraging people to use Tyk OAS!!). The purpose of this page changed when we went to Feature Maturity - I should have removed those lines then.

**Review by dcs3spp — APPROVED** [review]
Thanks @andyo-tyk PR LGTM, I'll merge and release to 5.4 and 5.5 docs

**Pull request merged**